src/net40/Radical.Windows.Presentation/Services/ConventionsHanlder.cs

```diff
@@ -78,11 +78,21 @@ public ConventionsHanlder(IMessageBroker broker, IReleaseComponents releaser, Bo
                         }
 
                         this.releaser.Release(vm);
+                        var disposableViewModel = vm as IDisposable;
+                        if( disposableViewModel != null)
+                        {
+                            disposableViewModel.Dispose();
+                        }
                     }
 
                     this.DetachViewBehaviors(view);
 
                     this.releaser.Release(view);
```

Comment:
Now it's clear also to me :smile:  now It's obvious why you're eventually calling `Dispose` two times, the first one because of the container release implementation (like castle that does the dispose), instead other containers (like Autofac) that does not support releasing/disposing of object, we dispose to the rescue.